File: Java/Day2/Complex.java
class Complex{
    
    private double real;
    private double imag;
    
    public Complex(double r, double i){
        real = r;
        imag = i;
    }
    
    public Complex add(double r, double i){
        double sumReal = real + r;
        double sumImag = imag + i;
        
        return new Complex(sumReal, sumImag);
    }
    
    public Complex subtract(double r, double i){
        double subReal = Math.abs(real - r);
        double subImag = Math.abs(imag - i);
        
        return new Complex(subReal, subImag);
    }
    
    public String toString(){
        return real + " + " + imag + "i";
    }
}

File: Kotlin/Day3/src/Complex.kt
class Complex(
    val real: Double,
    val imag: Double = 0.0
) {

    constructor(real: Double) : this(real, 0.0)

    operator fun plus(other: Complex): Complex {
        return Complex(
            real = this.real + other.real,
            imag = this.imag + other.imag
        )
    }

    operator fun minus(other: Complex): Complex {
        return Complex(
            real = this.real - other.real,
            imag = this.imag - other.imag
        )
    }
}

fun Complex.myPrint() {
    val sign = if (imag >= 0) "+" else "-"
    println("$real $sign ${kotlin.math.abs(imag)}i")
}

fun main() {

    val c1 = Complex(3.0, 4.0)
    val c2 = Complex(real = 1.0)

    val sum = c1 + c2
    val diff = c1 - c2

    sum.myPrint()
    diff.myPrint()
}

